crates/holochain_p2p/src/spawn/actor.rs

```diff
@@ -192,18 +198,178 @@ impl event::HcP2pHandler for WrapEvtSender {
     }
 }
 
-#[derive(Debug)]
+type Respond = tokio::sync::oneshot::Sender<crate::wire::WireMessage>;
+
+struct Pending {
+    this: Weak<Mutex<Self>>,
+    map: HashMap<u64, Respond>,
```

Comment:
Why do you keep track of the msg ids?

Reply:
When the response message comes in we relate the incoming msg_id to the one we included in the outgoing request, so we can hand the response back to the requesting process.

Reply:
Why wasn't that needed before?

Reply:
It was, it was just handled in legacy kitsune, and now kitsune has a simpler api and hcp2p takes on that complexity